xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/documentTags.vm

```diff
@@ -136,20 +136,14 @@
         #set($hasTagsPlugin = false)
         #set($tags = $doc.getTagList())
     #end
-    ##   #if($!tags.size() == 0)
-    ##     No tags
-    ##   #else
-    $services.localization.render('core.tags.list.label')
+    #if($hasedit || !$tags.isEmpty())
+        $services.localization.render('core.tags.list.label')
+    #end
     #foreach($tag in $tags)
         #displayTag($tag)
     #end
-    ##   #end
-    #if($hasedit)
-        #if($xwiki.tag)
-          <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
-        #else
-            ## TODO
-        #end
+    #if($hasedit && $xwiki.tag)
+        <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
```

Comment:
This line is too long and looks easy to split to make it more readable. Don't forget to use `##` at the end of lines to prevents unwanted line break in the resulting html.

Reply:
I agree, but imho it's fine to keep it as-is as this pull request didn't change anything in this line (apart from the indentation). So, yes, would be a nice improvement but imho not necessary as part of this pull request (but it also doesn't hurt).

Reply:
I agree, not mandatory at all.
I mainly raised this comment because I had a hard time understanding if something had changed, because of the too long line.
So changing this is mainly to be nice to future maintainers of this code :)